docs/react-native/v2/features/change-role.mdx

```diff
@@ -3,23 +3,58 @@ title: Change Role
+
+
+2. Check what actions this Role can perform. i.e can it change someone else's current Role, End the Meeting, or remove someone from the room. This is can be discovered by checking the `permissions` property.
+
+
+In certain scenarios, you may want to change someone's role.
+
+Imagine an audio room with 2 roles `speaker` and `listener`. Only someone with a `speaker` role can publish audio to the room while a `listener` can only subscribe.
+
+Now at some point, the `speaker` may decide to nominate some `listener` to become a `speaker`. 
+
+This is where Change Role capabilities come in play.
+
+
+You may choose to do either:
+
+1. [Single Peer Role Change](#single-peer-role-change): Change the role of a single peer to a specified one using the `changeRoleOfPeer` API
+
+
+2. [Bulk Role Change](#bulk-role-change): Change the role of all peers with a certain role, to a specified one using the `changeRoleOfPeersWithRoles` API
 
-```js
-const changeRole: boolean = peer.role.permissions.changeRole;
-```
 
-## Example
+## Single Peer Role Change
 
-Imagine an audio room with 2 roles "speaker" and "listener". Only someone with a "speaker" role can publish audio to the room while "listener" can only subscribe. Now at some point "speaker" may decide to nominate some "listener" to become a "speaker". This is where the changeRole comes in.
+To invoke the API you will need 4 parameters.
 
-## Request Role Change
+-   `forPeer`: An instance of `HMSPeer` of the peer who's role you want to change.
 
-You can either request a user to accept a role change request or forcefully change user's current role. If the request is accepted, the role and permissions will get updated. If the request is denied and you still want to change user's role then use force flag.
+-   `toRole`: The `HMSRole` instance for the target role.
 
-Imagine that the newly nominated speaker is not behaving nicely and we want to move him back to listener without a prompt. This is where the force parameter comes in. When it is set to true the other party will not receive a confirmation roleChangeRequest but instead will straight away receive a new set of updated permissions and stop publishing.
+-   `force`: Whether you want to change their role without asking them or give them a chance to accept/reject.
+
+-   `hmsActionResultListener`: The `HMSActionResultListener` that will get a success or failure callback depending on the result of the request.
+
+Note: success doesn't mean that the role was changed, just that the server accepted the request as valid.
+
+
+
+All the peers that are in the current room are accessible via the `peers` property of the `HMSRoom` instance. 
+
+
+A list of all available roles in the current room can be accessed via the `getRoles` method of `HMSSDK`.
+
+
+Now with all the parameters you can invoke the `changeRoleOfPeer` API:
 
 ```js
 import { HMSRole, HMSRemotePeer } from '@100mslive/react-native-hms';
```

Comment:
Should be use `hmsInstance?.getRoles()` method instead of `hmsInstance?.knownRoles`?